app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -2,41 +2,69 @@ package com.lightricks.feedexercise.ui.feed
 
 import androidx.lifecycle.*
 import com.lightricks.feedexercise.data.FeedItem
+import com.lightricks.feedexercise.data.FeedRepository
+import com.lightricks.feedexercise.data.Repository
 import com.lightricks.feedexercise.util.Event
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.disposables.CompositeDisposable
+import io.reactivex.rxjava3.disposables.Disposable
 import java.lang.IllegalArgumentException
+import java.util.function.Supplier
 
 /**
  * This view model manages the data for [FeedFragment].
  */
-open class FeedViewModel : ViewModel() {
+class FeedViewModel(private val feedRepository: Repository) : ViewModel() {
     private val stateInternal: MutableLiveData<State> = MutableLiveData<State>(DEFAULT_STATE)
     private val networkErrorEvent = MutableLiveData<Event<String>>()
+    private val compositeDisposable = CompositeDisposable()
+    private var refreshFeedItemsDisposable: Disposable? = null
 
-    fun getIsLoading(): LiveData<Boolean> {
-        //todo: fix the implementation
-        return MutableLiveData()
+    init {
+        refresh()
+        observeFeedItems()
     }
 
-    fun getIsEmpty(): LiveData<Boolean> {
-        //todo: fix the implementation
-        return MutableLiveData()
+    fun refresh() {
+        updateState { copy(isLoading = true) }
+
+        if (!getState().isRefreshing) {
```

Comment:
`isLoading` and `isRefreshing` are taking track of the same operation so remove either one of the two.